crates/qasm3/src/build.rs

```diff
@@ -0,0 +1,406 @@
+    ) -> PyResult<()> {
+        let name_id = decl
+            .name()
+            .as_ref()
+            .map_err(|err| QASM3ImporterError::new_err(format!("internal error {:?}", err)))?;
+        let name_symbol = &ast_symbols[name_id];
+        match name_symbol.symbol_type() {
+            Type::Bit(is_const) => {
+                if is_const.clone().into() {
+                    Err(QASM3ImporterError::new_err("cannot handle consts"))
+                } else if decl.initializer().is_some() {
+                    Err(QASM3ImporterError::new_err(
+                        "cannot handle initialised bits",
+                    ))
+                } else {
+                    self.add_clbit(py, name_id.clone())
+                }
+            }
+            Type::BitArray(dims, is_const) => {
+                if is_const.clone().into() {
+                    Err(QASM3ImporterError::new_err("cannot handle consts"))
+                } else if decl.initializer().is_some() {
+                    Err(QASM3ImporterError::new_err(
+                        "cannot handle initialised registers",
+                    ))
+                } else {
+                    match dims {
+                        ArrayDims::D1(size) => {
+                            self.add_creg(py, name_id.clone(), name_symbol.name(), *size)
+                        }
+                        _ => Err(QASM3ImporterError::new_err(
+                            "cannot handle quantum registers with more than one dimension",
+                        )),
+                    }
+                }
+            }
+            ty => Err(QASM3ImporterError::new_err(format!(
+                "unhandled classical type: {:?}",
+                ty,
+            ))),
+        }
+    }
+
+    fn declare_quantum(
+        &mut self,
+        py: Python,
+        ast_symbols: &SymbolTable,
+        decl: &asg::DeclareQuantum,
+    ) -> PyResult<()> {
+        let name_id = decl
+            .name()
+            .as_ref()
+            .map_err(|err| QASM3ImporterError::new_err(format!("internal error {:?}", err)))?;
+        let name_symbol = &ast_symbols[name_id];
+        match name_symbol.symbol_type() {
+            Type::Qubit => self.add_qubit(py, name_id.clone()),
+            Type::QubitArray(dims) => match dims {
+                ArrayDims::D1(size) => {
+                    self.add_qreg(py, name_id.clone(), name_symbol.name(), *size)
+                }
+                _ => Err(QASM3ImporterError::new_err(
+                    "cannot handle quantum registers with more than one dimension",
+                )),
+            },
+            _ => unreachable!(),
+        }
+    }
+
+    fn call_gate(
+        &mut self,
+        py: Python,
+        ast_symbols: &SymbolTable,
+        call: &asg::GateCall,
+    ) -> PyResult<()> {
+        if call.modifier().is_some() {
+            return Err(QASM3ImporterError::new_err("gate modifiers not handled"));
```

Comment:
Is this more accurate? I'm assuming this is a current limitation and future releases might have support for gate modifiers:

```suggestion
            return Err(QASM3ImporterError::new_err("gate modifiers currently not handled"));
```

Reply:
I'll push up a commit handling them, now they're done in the parser.

Reply:
Done in 5b8398ab13

Reply:
Note that I started a PR in https://github.com/jakelishman/qiskit-terra/pull/23 . I did not know if you would have time to get to it.